main.js

```diff
@@ -61,7 +73,43 @@ myList.addEventListener("click", function (e) {
   }
 });
 
-//persisting the todos
+document.addEventListener("DOMContentLoaded", function () {
+  loadTodos();
+});
+
+function loadTodos() {
+  const savedTodos = localStorage.getItem("todos");
+  if (savedTodos) {
+    const todos = JSON.parse(savedTodos);
+    todos.forEach((todo) => {
+      const li = document.createElement("li");
+      li.textContent = todo.text;
+
+      let deleteButton = document.createElement("button");
+      deleteButton.textContent = "Delete";
+      deleteButton.classList.add("btn-delete");
+
+      let editButton = document.createElement("button");
+      editButton.textContent = "Edit";
+      editButton.classList.add("btn-edit");
+
+      li.appendChild(editButton);
+      li.appendChild(deleteButton);
+
+      if (todo.completed) {
+        li.classList.add("complete");
+      }
+
+      myList.appendChild(li);
+    });
+  }
+}
+
 function saveTodos() {
-  localStorage.setItem("todos", myList.innerHTML);
+  const todos = Array.from(myList.children).map((li) => ({
```

Comment:
Dividing the logic into a module with state and separate rendering will get rid of the additional mapping from HTML to JSON.